libopae/plugins/xfpga/metrics/metrics_max10.c

```
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
 			metric_type = FPGA_METRIC_TYPE_UNKNOWN;
 		}
 
-		result = get_metric_data_info(group_name, metric_name, fpga_max10_metric_metadata, MAX10_MDATA_SIZE, &metric_data);
-		if (result != FPGA_OK) {
-			FPGA_MSG("Failed to get metric metadata ");
-			if (tmp) {
-				free(tmp);
-			}
-			continue;
-		}
-
 		if (tmp) {
 			free(tmp);
 		}
 
+		// Metric Units
+		if (EOK == strstr_s(metric_name, strlen(metric_name),
+				POWER, strlen(POWER), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				POWER_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+				VOLTAGE, strlen(VOLTAGE), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				VOLTAGE_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+					CURRENT, strlen(CURRENT), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				CURRENT_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+				TEMPERATURE, strlen(TEMPERATURE), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				TEMPERATURE_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+					CLCOK, strlen(CLCOK), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				CLCOK_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```

Just continuing here silently ignores the error.